Improve Practice UX

menvae opened the pull request:
Aims to improve the user experience of the practice ui

## Changes
- practice controls can now be held instead of spamming the button
- add sound samples (same as slider) for auditory feedback
- add current bar highlighting, shows which bar is part of the current played part
- add new playhead, complements the bar highlighting, shows where exactly is playing in the song
- add new draggable bars, pressing buttons can be tedious so this is a big QoL improvement

## Code Changes
- add new ``FooterPracticeRangeController`` (has a private class ``RangePoint``) & ``FooterPracticePlayhead``
- current bar highlighting in ``FooterPracticeGraph`` is updated every frame

All components are seperate from each other
```cs
new Container
{
    RelativeSizeAxes = Axes.X,
    Height = 80,
    Children = new Drawable[]
    {
        new FooterPracticeGraph(start, end)
        {
            RelativeSizeAxes = Axes.X,
            Height = 80,
        },
        new FooterPracticePlayhead(start, end)
        {
            RelativeSizeAxes = Axes.X,
            Height = 85,
        },
        new FooterPracticeRangeController(start, end)
        {
            RelativeSizeAxes = Axes.X,
            Height = 60,
            Margin = new MarginPadding { Bottom = 10, Top = 10 }
        }
    }
},
```

menvae committed:
allow holding practice controls for fast seeking

menvae committed:
Merge branch 'InventiveRhythm:main' into practice-ux

menvae committed:
resolve conflict

menvae committed:
update current bar color based on current time

menvae committed:
add playhead to practice timeline
also fixesthe bars

menvae committed:
add range points to practice
also remove unused imports

menvae committed:
add samples to new practice

menvae committed:
right click to reset practice range

menvae committed:
improve look of practice drag bar

menvae committed:
better resetting for drag in practice

menvae committed:
significantly improve consistency of drag bars

menvae committed:
I believe this is better

flustix commented:
Can't even test this.

```cs
2025-07-29 20:15:26 [error]: System.NullReferenceException: Object reference not set to an instance of an object.
2025-07-29 20:15:26 [error]: at fluXis.Screens.Select.Footer.Practice.FooterPractice.load() in Y:\TeamFluXis\fluXis\fluXis\Screens\Select\Footer\Practice\FooterPractice.cs:line 44
2025-07-29 20:15:26 [error]: at InvokeStub_FooterPractice.load(Object, Object, IntPtr*)
2025-07-29 20:15:26 [error]: at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
```

menvae committed:
fix range controller initlization

menvae commented:
> Can't even test this.

Made a HUGE oversight, I was trying to access the graph's members before they got initialized...

menvae committed:
better start width for range controller

flustix commented:
Also I don't think the line for the current time is needed, the highlighted bar seems to function the same.

menvae commented:
> Also I don't think the line for the current time is needed, the highlighted bar seems to function the same.

Well I thought that having the playhead would be nice for longer songs since it allows you to see more accurately, was also thinking of making it seek by dragging it but that might be out of scope or maybe not a good idea because it might conflict with the drag bars.
Either way I'm on board of removing it if needed,

---

> Handle can't be dragged further but the time goes below that.
> https://github.com/user-attachments/assets/a0315791-ba4e-4274-b456-4f6d3ba91ac8
>
> This also doesn't feel right
> https://github.com/user-attachments/assets/4e995403-1122-463f-91ac-9f506542a2f4

I can't access the links

menvae commented:
> Handle can't be dragged further but the time goes below that.
> https://github.com/user-attachments/assets/a0315791-ba4e-4274-b456-4f6d3ba91ac8

hmm that's weird.. maybe that's because the drag bars not colliding is based on their position not time, or maybe because the right range point has a different origin from the other one

> This also doesn't feel right
> https://github.com/user-attachments/assets/4e995403-1122-463f-91ac-9f506542a2f4

yes I'm aware of this, it's on purpose
since the practice box is relatively small and there isn't a much of a big margin on the sides I thought it'd be annoying if it was deselected but, the better choice and (what you mostly like are asking) is that it only moves when the mouse is in the draggable area

menvae committed:
apply requested changes
removing unused imports last

menvae committed:
major improvement to drag range points

- more reliable and consistent and accurate
- better clamping, no longer is there a big gap at the edges although other points are still clamped to each other which is intentional
- no longer need max width now

menvae committed:
only drag range points within acceptable bounds

+ code style changes

menvae committed:
fix other range point bounds

I seriously did not expect removing brackets would cause such a racket

menvae committed:
remove playhead from practice

menvae committed:
Merge branch 'main' into practice-ux

menvae committed:
ignore Y bounds when dragging

menvae committed:
Merge branch 'main' into practice-ux

menvae committed:
rename points to bounds

+ make it more relative
+ make bounds public
+ make offset public
